sufia-models/app/actors/sufia/generic_file/actor.rb

```diff
@@ -102,11 +107,22 @@ def save_and_record_committer
         return false unless generic_file.save
       rescue RSolr::Error::Http => error
         ActiveFedora::Base.logger.warn "Sufia::GenericFile::Actor::save_and_record_committer Caught RSOLR error #{error.inspect}"
+        Sufia.config.upload_logger.warn "Sufia::GenericFile::Actor::save_and_record_committer Error saving #{generic_file.title}. Caught RSOLR error #{error.inspect}"
+        save_tries += 1
+        # fail for good if the tries is greater than 3
+        raise error if save_tries >= 3
+        sleep 0.01
+        retry
+      rescue Ldp::Gone => error
+        Sufia.config.upload_logger.warn "Sufia::GenericFile::Actor::save_and_record_committer Error saving #{generic_file.title}. Caught LDP error #{error.inspect}"
         save_tries += 1
         # fail for good if the tries is greater than 3
         raise error if save_tries >= 3
         sleep 0.01
         retry
+      rescue StandardError => error
+        Sufia.config.upload_logger.warn "Sufia::GenericFile::Actor::save_and_record_committer Error saving #{generic_file.title}. Caught Exception error #{error.inspect}"
```

Comment:
Sounds good, @elrayle 

I understand that there are different perspectives on logging represented in this PR, and I think this is a valid point of differing opinions. Therefore, I recommend that we merge this feature but make it optional and have the default be off; folks should opt in to using a secondary logger. Can you tweak the PR so that the default value of `upload_logger` is `nil`, and have the calls to `upload_logger.warn` check first to make sure the value of `upload_logger` isn't `nil`? How does that sound, @jcoyne @elrayle ?